File: test/java/PhylSowingDateCorrection.java
import package pheno;
 import java.util.List;
 import java.util.ArrayList;
 import java.math.BigDecimal;
public class PhylSowingDateCorrection
{
    public double FixPhyll;
    public PhylSowingDateCorrection(double _FixPhyll)
    {
        this.FixPhyll=_FixPhyll;
    }
}

class Estimation_PhylSowingDateCorrection
{
    public static PhylSowingDateCorrection CalculatePhylSowingDateCorrection(int SowingDay,double Latitude,int SDsa_sh,double Rp,int SDws,int SDsa_nh,double P)
    {


/*
     PhylSowingDateCorrection Model

    Author: 
    Reference: Modeling development phase in the 
                Wheat Simulation Model SiriusQuality.
                See documentation at http://www1.clermont.inra.fr/siriusquality/?page_id=427
    Instituton: INRA Montpellier
    Abstract: Correction of the Phyllochron Varietal parameter according to sowing date 
    
*/
        double FixPhyll;

        if (Latitude < 0)
        {
        	if (SowingDay > SDsa_sh)
        	{
        		FixPhyll = P * (1 - Rp * Math.min(SowingDay - SDsa_sh, SDws));
        	}
        	else FixPhyll = P;
        }
        else
        {
        	if (SowingDay < SDsa_nh)
        	{
        		FixPhyll = P * (1 - Rp * Math.min(SowingDay, SDws));
        	}
        	else FixPhyll = P;
        }
        return new PhylSowingDateCorrection(FixPhyll);
    }

}

File: src/cpp/source/PhylSowingDateCorrection.cpp
#include <iostream>
#include <vector>
#include<string>
#include<algorithm>
#include<cmath>
#include "PhylSowingDateCorrection.h"
PhylSowingDateCorrection Calculate_PhylSowingDateCorrection(int sowingDay,double latitude,int sDsa_sh,double rp,int sDws,int sDsa_nh,double p)
{
    double fixPhyll;
    if (latitude < 0)
    {
        if (sowingDay > sDsa_sh)
        {
            fixPhyll = p * (1 - rp * fmin(sowingDay - sDsa_sh, sDws));
        }
        else fixPhyll = p;
    }
    else
    {
        if (sowingDay < sDsa_nh)
        {
            fixPhyll = p * (1 - rp * fmin(sowingDay, sDws));
        }
        else fixPhyll = p;
    }

    return PhylSowingDateCorrection{fixPhyll};
}
